src/graph/xml.cc

```diff
@@ -636,15 +636,15 @@ ncclResult_t ncclTopoGetXmlFromGpu(struct ncclXmlNode* pciNode, uint32_t rocmDev
   int sm;
   NCCLCHECK(xmlGetAttrInt(gpuNode, "sm", &sm));
 
-  int gcn;
+  const char* gcn;
   NCCLCHECK(xmlGetAttrIndex(gpuNode, "gcn", &index));
   if (index == -1) {
     hipDeviceProp_t devProp;
     CUDACHECK(hipGetDeviceProperties(&devProp, 0));
-    gcn = devProp.gcnArch;
-    NCCLCHECK(xmlSetAttrInt(gpuNode, "gcn", gcn));
+    gcn = devProp.gcnArchName;
```

Comment:
To avoid RCCL build from source getting broken on older ROCm releases, we need to use HIP_NO_GCNARCHNAME to switch between gcnArch and gcnArchName.

Reply:
had this in the [previous commit](https://github.com/ROCmSoftwarePlatform/rccl/commit/fea5d1ce3eb5584dc40387874b0454526599d554)... should we still keep it given that gcnArch was deprecated in ROCm 3.7?